provisioning/service/src/Models/QueriedPage.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Net.Http;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.Devices.Provisioning.Service
+{
+    /// <summary>
+    /// Contains the result of a twin, scheduled job or raw query.
+    /// </summary>
+    internal sealed class QueriedPage<T>
+    {
+        private const string ContinuationTokenHeader = "x-ms-continuation";
+
+        // Payload is taken separately from http response because reading the payload should only be done
+        // in an async function.
+        internal QueriedPage(HttpResponseMessage response, string payload)
+        {
+            Items = JsonConvert.DeserializeObject<IReadOnlyList<T>>(payload);
+            ContinuationToken = response.Headers.SafeGetValue(ContinuationTokenHeader);
+        }
+
+        [JsonProperty("items")]
```

Comment:
For these to work to deserialize, you'll need a parameterless ctor. Is this class only used for serializing?

Reply:
This class isn't used for serialization, so I'll get rid of these Json property tags